BlueDBOnDisk/src/main/java/io/bluedb/disk/BlueDbOnDisk.java

```diff
@@ -2,28 +2,36 @@
 
 import java.io.Serializable;
 import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.Map;
 
 import io.bluedb.api.BlueCollection;
 import io.bluedb.api.BlueDb;
 import io.bluedb.api.exceptions.BlueDbException;
 import io.bluedb.disk.collection.BlueCollectionImpl;
-import io.bluedb.disk.serialization.BlueSerializer;
-import io.bluedb.disk.serialization.ThreadLocalFstSerializer;
 
 public class BlueDbOnDisk implements BlueDb {
 
 	private final Path path;
-	private BlueSerializer serializer;
+	
+	private final Map<String, BlueCollection<?>> collections = new HashMap<>();
 	
 	BlueDbOnDisk(Path path, Class<?>...registeredSerializableClasses) {
 		this.path = path;
-		this.serializer = new ThreadLocalFstSerializer(registeredSerializableClasses);
 	}
 
 	@Override
-	public <T extends Serializable> BlueCollection<T> getCollection(Class<T> type, String name) {
-		// TODO make sure only one Collection object per collection to avoid concurrency issues
-		return new BlueCollectionImpl<>(this, type);
+	@SuppressWarnings("unchecked")
+	public <T extends Serializable> BlueCollection<T> getCollection(Class<T> type, String name) throws BlueDbException {
+		synchronized (collections) {
+			String key = type.getName() + "-" + name;
```

Comment:
I made the type and name be the key of the collection. This means that if they give us the same name for different types it will still work. This isn't really what we discussed but is there a problem with it? I guess there would be a collision on the directories... I'll change this.